embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/internal/logs/LogOrchestratorTest.kt

```diff
@@ -0,0 +1,110 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.concurrency.BlockingScheduledExecutorService
+import io.embrace.android.embracesdk.fakes.FakeClock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+
+internal class LogOrchestratorTest {
+
+    companion object {
+        private const val now = 123L
+    }
+
+    private lateinit var logOrchestrator: LogOrchestrator
+    private lateinit var onLogsStored: () -> Unit
+
+    private lateinit var executorService: BlockingScheduledExecutorService
+    private lateinit var scheduledWorker: ScheduledWorker
+    private val clock = FakeClock()
+    private val logSink: LogSinkImpl = mockk(relaxed = true)
+
+    @Before
+    fun setUp() {
+        executorService = BlockingScheduledExecutorService()
+        scheduledWorker = ScheduledWorker(executorService)
+        clock.setCurrentTime(now)
+        val onLogsStoredSlot = slot<() -> Unit>()
+        every {
+            logSink.callOnLogsStored(capture(onLogsStoredSlot))
+        } returns Unit
+        logOrchestrator = LogOrchestrator(scheduledWorker, clock, logSink)
+        onLogsStored = onLogsStoredSlot.captured
+    }
+
+    @Test
+    fun `a listener is set on initialization`() {
+        assertNotNull(onLogsStored)
+    }
+
+    @Test
+    fun `send a batch of logs when the batch max size is reached`() {
+        val logs = mutableListOf<EmbraceLogRecordData>()
+        every { logSink.completedLogs() } returns logs
+
+        // Fill the sink with max batch size - 1 logs
+        repeat(49) {
+            logs.add(mockk())
```

Comment:
We can probably get a similar unit test without using mocks & `verify`, by either using real objects, or creating fake objects. These are substantially easier to refactor compared to mocks & are less likely to break when something changes.